weather/src/main/java/bot/WeatherBot.java

```diff
@@ -35,17 +24,30 @@ public static void main(String[] args) {
             LOGGER.error("Telegram bot cannot be launched. Token is undefined.");
             System.exit(0);
         }
+        final String TOKEN = args[0];
+
         if (args.length == 1) {
             LOGGER.error("Telegram bot cannot work correctly. Access key for weatherstack.com is undefined.");
             System.exit(0);
         }
+        final String API_ACCESS_KEY = args[1];
+
+        // if database is available, try to find weather info in cache first
+        final WeatherClient weatherClient;
+
+        if (args.length == 3) {
+            final String DATABASE_PASS = args[2];
+            weatherClient = new WeatherCache(new WeatherApiClient(API_ACCESS_KEY), DATABASE_PASS);
+        } else {
+            LOGGER.warn("Database access password is undefined. Connection to the database is impossible.");
+            LOGGER.info("Weather forecast search will be executed only through API weatherstack.com");
+            weatherClient = new WeatherApiClient(API_ACCESS_KEY);
+        }
 
-        final String token = args[0];
-        final String accessKey = args[1];
-
-        TelegramBot bot = new TelegramBot(token);
+        TelegramBot bot = new TelegramBot(TOKEN);
         LOGGER.info("Telegram bot " + NAME + " started");
 
+        final WeatherClient finalWeatherClient = weatherClient;
```

Comment:
Not used, it's enough to have `weatherClient`.

Reply:
sorry, accidental variable)